codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -0,0 +1,642 @@
+                pattern.append("(?P<${it.content}>")
+                if (it.isGreedyLabel) {
+                    pattern.append(".+")
+                } else {
+                    pattern.append("[^/]+")
+                }
+                pattern.append(")")
+            } else {
+                pattern.append(it.content)
+            }
+        }
+        val errorShape = operationShape.errorSymbol(symbolProvider)
+        with(writer) {
+            rustTemplate(
+                """
+                    #{LazyStatic}::lazy_static! {
+                        static ref RE: #{Regex}::Regex = #{Regex}::Regex::new("$pattern").unwrap();
+                    }
+                """.trimIndent(),
+                *codegenScope,
+            )
+            rustBlock("if let Some(captures) = RE.captures(request.uri().path())") {
+                pathBindings.forEach {
+                    val deserializer = generateParseLabelFn(it)
+                    rustTemplate(
+                        """
+                            if let Some(m) = captures.name("${it.locationName}") {
+                                input = input.${it.member.setterName()}(
+                                    #{deserializer}(m.as_str())?
+                                );
+                            }
+                        """.trimIndent(),
+                        "deserializer" to deserializer,
+                        "E" to errorShape,
+                    )
+                }
+            }
+        }
+    }
+
+    private fun serverRenderHeaderParser(writer: RustWriter, binding: HttpBindingDescriptor, operationShape: OperationShape) {
+        val httpBindingGenerator =
+            ResponseBindingGenerator(
+                ServerRestJson(codegenContext),
+                codegenContext,
+                operationShape,
+            )
+        val deserializer = httpBindingGenerator.generateDeserializeHeaderFn(binding)
+        writer.rust(
+            """
+                #T(request.headers())?
+            """.trimIndent(),
+            deserializer,
+        )
+    }
+
+    private fun generateParseLabelFn(binding: HttpBindingDescriptor): RuntimeType {
+        check(binding.location == HttpLocation.LABEL)
+        val target = model.expectShape(binding.member.target)
+        return when {
+            target.isStringShape -> generateParseLabelStringFn(binding)
+            target.isTimestampShape -> generateParseLabelTimestampFn(binding)
+            else -> generateParseLabelPrimitiveFn(binding)
+        }
+    }
+
+    private fun generateParseLabelStringFn(binding: HttpBindingDescriptor): RuntimeType {
+        val output = symbolProvider.toSymbol(binding.member)
+        val fnName = generateParseLabelFnName(binding)
+        return RuntimeType.forInlineFun(fnName, operationDeserModule) { writer ->
+            writer.rustBlockTemplate(
+                "##[allow(dead_code)] pub fn $fnName(value: &str) -> std::result::Result<#{O}, #{Error}>",
```

Comment:
Why not insert above:

```
Attribute.AllowUnused.render(it)
```

where `Attribute` is defined in `RustTypes.kt`.

Same comment goes for all other uses of `#[allow(dead_code)]` in this file.

Reply:
Yeah, it's definitely better.

Reply:
(but also, I'm a little suspicious why this code is dead)

Reply:
I think you are right @rcoh.. This code _is_not_ dead. I am removing the markers. This was a leftover from the old RestJson1 implementation